### PR TITLE
mssw chapter 7

### PR DESCRIPTION
**Pull request by Spinodal**

### CODE REVIEW

File: chapter07/mssw/exercise1.pl [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        $dow += 7;
+    }
+}
+my $stop = $start + $seconds_per_day;
+
+my ( $gather, $yield ) = gather_mtime_between( $start, $stop );
+find( $gather, @starting_directories );
+
+# my @files = $yield->();   # why not reference?!
+my $files = $yield->();
```
Comment (dickyince) [inline chapter07/mssw/exercise1.pl:32]: Is this change reviewed anywhere :-P

File: chapter07/mssw/exercise1.pl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    return (
+        sub {
+            if ( -f $_ ) {
+                my $filename  = $File::Find::name;
+                my $mtime     = ( stat $_ )[9];
+                my $nice_time = gmtime($mtime);
+                if ( ( $mtime > $start ) && ( $mtime < $stop ) ) {
+
+                    # push @files, $filename; # bring on the hash!
+                    $files{$filename} = $nice_time;
```
Comment (dickyince) [inline chapter07/mssw/exercise1.pl:56]: funny looking list you have here

File: chapter07/mssw/exercise1.pl [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## for my $file (@$files) { # reference ftw!
+#foreach my $file (@$files){ # **** that last reference, i want a hash now!
+#    my $mtime = ( stat $file )[9];    # mtime via slice
+#    my $when  = localtime $mtime;
+#    print "$when: $file\n";
+#}
+foreach my $file ( keys %$files ) {
+    print "$files->{$file}: $file\n";
+}
```
Comment (dickyince) [inline chapter07/mssw/exercise1.pl:42]: again you have made up your own tune, not sure how this will go down with the rest of the band

File: chapter07/mssw/exercise1.pl [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+sub gather_mtime_between {
+    my ( $start, $stop ) = @_;
+    my %files = ();
+    return (
+        sub {
+            if ( -f $_ ) {
```
Comment (dickyince) [inline chapter07/mssw/exercise1.pl:49]: I don't think this is necessary as you have got the file name from File::Find and I'm sure it wouldn't give you a file that doesn't exist.

(que red/ajd)

File: chapter07/mssw/exercise1.pl [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                my $filename  = $File::Find::name;
+                my $mtime     = ( stat $_ )[9];
+                my $nice_time = gmtime($mtime);
```
Comment (dickyince) [inline chapter07/mssw/exercise1.pl:52]: SCOPE CREEP!!!!

File: chapter07/mssw/exercise1.pl (hunk 3 above)
Comment (utopia-2) [inline chapter07/mssw/exercise1.pl:42]: The rest of the band is totally okay with this. (Don't worry @Spinodal, I've got your back). :dragon:

File: chapter07/mssw/exercise1.pl (hunk 4 above)
Comment (alexdeas) [inline chapter07/mssw/exercise1.pl:49]: This way it'll skip directories, which you may or may not want to do.

File: chapter07/mssw/exercise1.pl (hunk 3 above)
Comment (ardavey) [inline chapter07/mssw/exercise1.pl:42]: @Utopia1138's views are not representative of those of the whole band.

File: chapter07/mssw/exercise1.pl [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                if ( ( $mtime > $start ) && ( $mtime < $stop ) ) {
```
Comment (ardavey) [inline chapter07/mssw/exercise1.pl:53]: What if the file was created exactly at either the start or stop time?

File: chapter07/mssw/exercise1.pl (hunk 4 above)
Comment (dickyince) [inline chapter07/mssw/exercise1.pl:49]: Good point

File: chapter07/mssw/exercise1.pl (hunk 6 above)
Comment (Spinodal) [inline chapter07/mssw/exercise1.pl:53]: yea probably should have done `($mtime >= $start)` if this was to run on a cron job every timeframe
